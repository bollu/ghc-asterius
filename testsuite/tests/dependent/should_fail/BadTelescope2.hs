{-# LANGUAGE DataKinds, PolyKinds, ExplicitForAll #-}

module BadTelescope2 where

import Data.Kind
import Data.Proxy

data SameKind :: k -> k -> *

foo :: forall a k (b :: k). SameKind a b
foo = undefined

bar :: forall a k (b :: k) (c :: Proxy b) (d :: Proxy a). Proxy c -> SameKind b d
bar = undefined
