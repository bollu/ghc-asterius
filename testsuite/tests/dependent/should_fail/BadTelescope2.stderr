
BadTelescope2.hs:10:8: error:
    • These kind and type variables: a k (b :: k)
      are out of dependency order. Perhaps try this ordering:
        k (a :: k) (b :: k)
    • In the type signature: foo :: forall a k (b :: k). SameKind a b

BadTelescope2.hs:13:81: error:
    • Expected kind ‘k’, but ‘d’ has kind ‘Proxy a’
    • In the second argument of ‘SameKind’, namely ‘d’
      In the type signature:
        bar :: forall a k (b :: k) (c :: Proxy b) (d :: Proxy a).
               Proxy c -> SameKind b d
